localstack/utils/docker.py

```diff
@@ -0,0 +1,401 @@
+import json
+import logging
+import os
+import subprocess
+from enum import Enum, unique
+from typing import Dict, List, Optional, Tuple, Union
+
+from localstack import config
+from localstack.utils.bootstrap import PortMappings
+from localstack.utils.common import TMP_FILES, rm_rf, safe_run, save_file, short_uid
+
+LOG = logging.getLogger(__name__)
+
+
+@unique
+class DockerContainerStatus(Enum):
+    DOWN = -1
+    NOT_EXISTENT = 0
+    UP = 1
+
+
+class ContainerException(Exception):
+    def __init__(self, message, stdout, stderr) -> None:
+        self.message = message
+        self.stdout = stdout
+        self.stderr = stderr
+
+
+class NoSuchContainer(ContainerException):
+    def __init__(self, message, container_name_or_id, stdout=None, stderr=None) -> None:
+        super().__init__(message, stdout, stderr)
+        self.container_name_or_id = container_name_or_id
+
+
+class NoSuchImage(ContainerException):
+    def __init__(self, message, image_name, stdout=None, stderr=None) -> None:
```

Comment:
nit: we could probably make the `message` argument optional here:
```
     def __init__(self, image_name, message=None, stdout=None, stderr=None):
         message = message or f'Docker image {image_name} not found'
         super().__init__(message, stdout, stderr)
```

Also, not sure if the type hint `-> None` adds much value for the constructors here?

Reply:
Value, no, it was actually autogenerated by VS Code when creating the constructor.